packages/scene-composer/src/components/toolbars/common/ToolbarItem.tsx

```diff
@@ -103,13 +26,18 @@ export function ToolbarItem<T extends ToolbarItemOptions>({
   const itemContainerRef = useRef<HTMLDivElement | null>(null);
 
   function handleItemClick(item: T) {
+    type === 'mode-select' && setSelectedItem(item);
     onClick && onClick(item);
   }
 
   useEffect(() => {
     initialSelectedItem && setSelectedItem(initialSelectedItem);
   }, [initialSelectedItem]);
 
+  useEffect(() => {
+    !Array.isArray(items) && setSelectedItem(items);
```

Comment:
why is `items` also an `item`? I don't follow why sometimes `items` is not an array?

Reply:
when there is only one item, it will not be an array. this is also legacy code... I can maybe change it to only be an array..

Reply:
should definitely avoid overloading `items` to also be `item` but understand that its outside of the scope of the PR